### PR TITLE
Added possibility to connect to a MQTT server with a client certificate

### PR DESCRIPTION
**Pull request by Martinius**
I have added the possibility to specify a client key and certificate, which can be uses to connect to a MQTT Server. This connection is only attempted if both key and certificate are supplied. Right now this are two separate input parameters in the config.  The old option where just a server CA certificate was supplied still works.

### CODE REVIEW

**Comment by mwarning** [comment]
Hi there. Thank you very much.

Have you been able to test this? If yes, how does your test setup look like?

**Comment by mwarning** [comment]
The GUI would probably need some rework. Let's see tomorrow.

**Comment by Martinius** [comment]
I have tested this with the mosquitto mqtt broker. This is the config I used:

#Global
per_listener_settings true

#Certificate listener
listener 8883
cafile /etc/mosquitto/certs/ca.crt
certfile /etc/mosquitto/certs/basestation.crt
keyfile /etc/mosquitto/certs/basestation.key
require_certificate true

#Security
allow_anonymous false
password_file /etc/mosquitto/passwords.txt

I was able to connect and publish/receive telegrams. If you want to test it yourself here is a link to the tutorial I used to setup the broker and the certificates:
https://www.onetransistor.eu/2019/05/mosquitto-mqtt-tls-certificate.html

File: app/src/main/java/app/trigger/mqtt/MqttRequestHandler.java [hunk 1]
```diff
@@ -107,9 +111,17 @@ public void run() {
             if (address.startsWith("ssl://")) {
                 if (setup.certificate != null) {
                     // use given certificate only
-                    opts.setSocketFactory(
-                            Utils.getSocketFactoryWithCertificate(setup.certificate)
-                    );
+                    if(setup.client_certificate != null && setup.client_key != null) {
+                        PrivateKey client_private_key = PubkeyUtils.decodePrivate(setup.client_key.getPrivateKey(),"RSA");
```
Comment (mwarning) [inline app/src/main/java/app/trigger/mqtt/MqttRequestHandler.java:115]: We could try to use setup.client_key.getType() instead of "RSA".

**Comment by Martinius** [comment]
Yes this works.

**Comment by mwarning** [comment]
I need to adapt the GUi first before merging. But my mosqitto setup does not like the give certifiactes.

**Comment by Martinius** [comment]
I use MQTT Explorer to inspect what is happening on the Broker. It works fine with the certificates. If you have problems with the certificate and key generation I can recommend this script:
https://github.com/owntracks/tools/blob/master/TLS/generate-CA.sh

Best regards

**Comment by mwarning** [comment]
I've add the MR to master and did some refactoring afterwards.
My MQTT setup does not work yet, so MQTT support might be broken right now.

**Pull request closed without merging**